Sources/MapboxNavigation/NavigationViewController.swift

```diff
@@ -343,19 +345,47 @@ open class NavigationViewController: UIViewController, NavigationStatusPresenter
                                navigationOptions: navigationOptions)
     }
     
+    /**
+     Initializes a `NavigationViewController` that presents the user interface for following a predefined route based on the given options.
+     The route may come directly from the completion handler of the [MapboxDirections](https://docs.mapbox.com/ios/api/directions/) framework’s `Directions.calculate(_:completionHandler:)` method, MapboxCoreNavigation `MapboxRoutingProvider.calculateRoutes(options:completionHandler:)`, or it may be unarchived or created from a JSON object.
+     
+     - parameter routeResponse: `IndexedRouteResponse` object, containing selection of routes to follow.
+     - parameter navigationOptions: The navigation options to use for the navigation session.
+     */
+    required public init(for indexedRouteResponse: IndexedRouteResponse, navigationOptions: NavigationOptions? = nil) {
+        super.init(nibName: nil, bundle: nil)
+        
+        _ = prepareViewLoading(indexedRouteResponse: indexedRouteResponse,
+                               navigationOptions: navigationOptions)
+    }
+    
+    /**
+     Initializes a `NavigationViewController` that presents the user interface for following a predefined route based on the given options.
+     The route may come directly from the completion handler of the [MapboxDirections](https://docs.mapbox.com/ios/api/directions/) framework’s `Directions.calculateRoutes(matching:completionHandler:)` method, MapboxCoreNavigation `MapboxRoutingProvider.calculateRoutes(options:completionHandler:)`, or it may be unarchived or created from a JSON object.
+     
+     - parameter mapMatchingResponse: `MapMatchingResponse` object, containing selection of routes to follow.
+     - parameter routeIndex: The index of the route within the original `RouteResponse` object.
+     - parameter navigationOptions: The navigation options to use for the navigation session.
+     */
+    required public convenience init(for mapMatchingResponse: MapMatchingResponse, routeIndex: Int, navigationOptions: NavigationOptions? = nil) throws {
```

Comment:
Added this as a convenience init which will do setup and converting types as needed. Did not propagate `MapMatchingResponse` to router and navigation service as it is a non-preferable navigation source.